### PR TITLE
feat: Extract `tags` in `getTypeDocumentation`

### PR DESCRIPTION
**Pull request by nihgwu**
This PR added `jsDocTags` to propType, which will be very useful from component API migration, updated the show case how to use it

This is my first time to use ts-morph, feel free to edit the PR 😄

### CODE REVIEW

**Comment by souporserious** [comment]
Hey @nihgwu 👋 thank you for the PR! I'm in the middle of a bit of work right now, but will take a look at this as soon as I get a chance. From a quick overview things mostly look good though! One thing I'll note is `getPropTypes` has been deprecated (sorry for not specifying this anywhere 😅) in favor of [getTypeDocumentation](https://github.com/souporserious/tsxmod/blob/main/packages/utils/src/types/getTypeDocumentation.ts). If you'd like to move this work over to the utility that'd be great, if not no worries and I can get to it when I get time to look at this more closely.

**Comment by nihgwu** [comment]
@souporserious no worries, I'll take a look later

File: site/app/page.tsx [hunk 1]
```diff
@@ -83,18 +95,49 @@ export function Dialog(props: {
...
+
   const appElement = getJsxElement(appSourceFile, 'App')
-  const buttonPropTypes = getPropTypes(buttonSourceFile.getVariableDeclaration('Button'))
-  
+  const buttonPropTypes = getPropTypes(
+    buttonSourceFile.getVariableDeclaration('Button')
+  )
+
   // Write codemod here
+  const deprecatedButtonProps = buttonPropTypes.filter((propType) =>
```
Comment (nihgwu) [inline site/app/page.tsx:112]: actually I think the current code mod is wrong, we should just analyse element prop types in `App.tsx` and migrate based on deprecation message, in this way we can migrate the whole app, but I don't know how to achieve it yet, super new to ts-morph and ts compiler, will check later

**Comment by nihgwu** [comment]
@souporserious I tried it in my local project, but the type doc is empty, but I can get tags and description via `signature.getJsDocTags()`

**Comment by souporserious** [comment]
I did a bit of refactoring to add better docs support all around to `getTypeDocumentation`. Analyzing JSDocs has been added here https://github.com/souporserious/tsxmod/commit/c2786af283162558010070042dd56f01a3e13f59 through a `getJsDocMetadata` utility and now provides the tags with their respective description. Links are not processed as this should be handled by the end user either through a Markdown or JSDoc processor.

**Pull request closed without merging**